### PR TITLE
need to check the sharedInformer synced or not

### PR DESCRIPTION
**Pull request by wenlxie**
Need to check the informer synced or not before return the list result

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

File: pkg/quota/generic/evaluator.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ func ListResourceUsingInformerFunc(f informers.SharedInformerFactory, resource s
 		if err != nil {
 			return nil, err
 		}
+		if !informer.Informer().HasSynced() {
```
Comment (resouer) [inline pkg/quota/generic/evaluator.go]: What will happen if no this check it here?

Comment (wenlxie) [inline pkg/quota/generic/evaluator.go]: The quota's used field may become 0, and quota limit  not take effect.

Comment (derekwaynecarr) [inline pkg/quota/generic/evaluator.go]: have you seen this happen in practice?

File: pkg/quota/generic/evaluator.go [hunk 2]
```diff
@@ -41,6 +41,9 @@ func ListResourceUsingInformerFunc(f informers.SharedInformerFactory, resource s
...
+			return nil, fmt.Errorf("Informer had not syncd yet")
```
Comment (derekwaynecarr) [inline pkg/quota/generic/evaluator.go]: nit: synced

File: pkg/quota/generic/evaluator.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/quota/generic/evaluator.go]: the change is fine, i am just curious if this actually has happened.  i thought we had checks elsewhere in the code that ensured all informers had been synced.

File: pkg/quota/generic/evaluator.go (hunk 2 above)
Comment (wenlxie) [inline pkg/quota/generic/evaluator.go]: @derekwaynecarr Changed

File: pkg/quota/generic/evaluator.go (hunk 1 above)
Comment (wenlxie) [inline pkg/quota/generic/evaluator.go]: Yes, The scenario is I created a rs and has 30 replicas, but the quota is just 10. 
So there will be only 10 pod that can be created. But after I restart all of the kube processes on the master node, I find that the  created pod increased to 20, then I restart  again, the pod number increased to 30. I watched the quota during the startup, the result shows  the used pod becomes to 0  during start period.  I can  reproduce this every time with 1.5, but can't reproduce with 1.6.   Anyway I think it's better to check this, even there are no issues left.

**Comment by wenlxie** [comment]
@derekwaynecarr @resouer Can this PR get approved? It seems there is something wrong with the fedoration-e2e-gce test.

**Comment by resouer** [comment]
This is in the scope of @derekwaynecarr, leave him to make decision.

**Comment by resouer** [comment]
And fedoration-e2e-gce test just keep failing these days, you can ignore it.

**Comment by wenlxie** [comment]
@resouer Thanks.

**Pull request closed without merging**